### PR TITLE
Remote KMS support

### PR DESCRIPTION
**Pull request by priyawadhwa**
Currently only works with GCP KMS, but hopefully the interface makes it easy enough to add other providers.

Right now looks like:

```
$ ./cosign generate-key-pair -kms gcpkms://projects/priya-wadhwa/locations/global/keyRings/cosign/cryptoKeys/cosignKey 
Key ring projects/priya-wadhwa/locations/global/keyRings/cosign already exists in GCP KMS, moving on to creating key.
Created key projects/priya-wadhwa/locations/global/keyRings/cosign/cryptoKeys/cosignKey in GCP KMS

$ TLOG=1 ./cosign sign --force-tlog -kms gcpkms://projects/priya-wadhwa/locations/global/keyRings/cosign/cryptoKeys/cosignKey -a kms=true gcr.io/priya-wadhwa/test
Pushing signature to: gcr.io/priya-wadhwa/test:sha256-74e4a68dfba6f40b01787a3876cc1be0fb1d9025c3567cf8367c659f2187234f.cosign
force uploading signature of private image to tlog
Sucessfully appended to transparency log:  https://api.rekor.dev /api/v1/log/entries/eaed60066cc2a748c2e230f2df3074a3c75f72959337e53a353cbc592cf50729

$ TLOG=1 ./cosign verify -kms gcpkms://projects/priya-wadhwa/locations/global/keyRings/cosign/cryptoKeys/cosignKey -a kms=true gcr.io/priya-wadhwa/test
The following checks were performed on these signatures:
  - The specified annotations were verified.
  - The cosign claims were validated
  - The claims were present in the transparency log
  - The signatures were verified against the specified public key
{"Critical":{"Identity":{"docker-reference":""},"Image":{"Docker-manifest-digest":"sha256:74e4a68dfba6f40b01787a3876cc1be0fb1d9025c3567cf8367c659f2187234f"},"Type":"cosign container signature"},"Optional":{"kms":"true"}}
```

Not sure if it's worth it to include the `-kms` flag with verify -- there's probably a good chance public keys will be published elsewhere & that's what people would verify against (especially since I haven't yet figured out how to make public keys in the KMS publicly available).

kinda closes https://github.com/sigstore/cosign/issues/45, except that we should probably have other KMS providers as well

### CODE REVIEW

File: cmd/cli/sign.go [hunk 1]
```diff
@@ -161,5 +165,28 @@ func SignCmd(ctx context.Context, keyPath string,
...
+func sign(ctx context.Context, img *remote.Descriptor, keyPath string, payload []byte, pf cosign.PassFunc) (signature []byte, publicKey *ecdsa.PublicKey, err error) {
+	pass, err := pf(false)
+	if err != nil {
+		return
+	}
+	kb, err := ioutil.ReadFile(keyPath)
+	if err != nil {
+		return
+	}
+	pk, err := cosign.LoadPrivateKey(kb, pass)
```
Comment (dlorenc) [inline cmd/cli/sign.go:180]: Nicely done with this entire function

**Review by dlorenc — COMMENTED** [review]
This whole PR is amazing.

**Comment by dlorenc** [comment]
LGTM! want to squash?

**Comment by priyawadhwa** [comment]
Yup!

**Pull request merged**